src/lib/stores/decrypt.ts

```diff
@@ -0,0 +1,45 @@
+import { browser } from '$app/env';
+import { writable } from 'svelte/store';
+
+export const decryptPort = persistStore('port', null);
+
+// An enum or string union may be more appropriate here so we dont' get a
+// if thing and the falsiness evaluates to the null case, but start with this
+export const lastDecryptStatus = writable<null | boolean>(null);
+
+export const portNumber = writable<null | number>(null);
+
+export function lastDecryptFailure() {
```

Comment:
Do you feel like the name of this function sounds like a property? But then it's really a setter?